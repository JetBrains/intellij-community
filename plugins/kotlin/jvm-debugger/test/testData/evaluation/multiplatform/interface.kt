// MODULE: common
// FILE: common.kt

expect interface Platform {
    open fun name1(): String
    open fun name2(param: String = "name2"): String
}

// MODULE: jvm(common)
// FILE: jvm.kt


actual interface Platform {
    actual fun name1() = "name1"
    actual fun name2(param: String): String = param
}

class TestPlatform() : Platform


fun main() {
    // EXPRESSION: TestPlatform()
    // RESULT: instance of TestPlatform(id=ID): LTestPlatform;
    //Breakpoint!
    val a = 0

    // EXPRESSION: TestPlatform().name1()
    // RESULT: "name1": Ljava/lang/String;
    //Breakpoint!
    val b = 0

    // EXPRESSION: TestPlatform().name2()
    // RESULT: "name2": Ljava/lang/String;
    //Breakpoint!
    val c = 0

    // EXPRESSION: TestPlatform().name2("special")
    // RESULT: "special": Ljava/lang/String;
    //Breakpoint!
    val d = 0

}