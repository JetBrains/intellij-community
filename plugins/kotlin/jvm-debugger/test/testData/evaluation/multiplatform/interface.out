LineBreakpoint created at jvm.kt:17
LineBreakpoint created at jvm.kt:22
LineBreakpoint created at jvm.kt:27
LineBreakpoint created at jvm.kt:32
Run Java
Connected to the target VM
jvm.kt:17
Compile bytecode for TestPlatform()
jvm.kt:22
Compile bytecode for TestPlatform().name1()
jvm.kt:27
Compile bytecode for TestPlatform().name2()
jvm.kt:32
Compile bytecode for TestPlatform().name2("special")
Disconnected from the target VM

Process finished with exit code 0
